frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -101,43 +102,51 @@ export const savedInsightsLogic = kea<savedInsightsLogicType>([
                     basic: true,
                 }
 
-                const response = await api.get(
+                const legacyResponse: CountedPaginatedResponse<InsightModel> = await api.get(
```

Comment:
Yep, you're right - this will be a query based insight if it is already stored with query on the backend. `legacyInsight` in this context means an insight that is potentially a legacy one. The conversion function will give precedence to a `query` already present on an insight.

Reply:
Alright, good to know.